File: hard/15_knuth_morris_pratt_algorithm/knuth_morris_pratt_algorithm_test.py
import unittest
from knuth_morris_pratt_algorithm import knuthMorrisPrattAlgorithm

class TestProgram(unittest.TestCase):
    def test_case_1(self):
        self.assertEqual(knuthMorrisPrattAlgorithm("aefoaefcdaefcdaed", "aefcdaed"), True)

File: hard/15_knuth_morris_pratt_algorithm/knuth_morris_pratt_algorithm_test.go
package main

import (
	"reflect"
	"testing"
)

func TestKnuthMorrisPrattAlgorithm(t *testing.T) {
	testCases := []struct {
		str    string
		substr string
		want   bool
	}{
		{"aefoaefcdaefcdaed", "aefcdaed", true},
	}
	for _, tc := range testCases {
		got := knuthMorrisPrattAlgorithm(tc.str, tc.substr)
		if !reflect.DeepEqual(got, tc.want) {
			t.Errorf("knuthMorrisPrattAlgorithm(%v, %v) = %v; want %v", tc.str, tc.substr, got, tc.want)
		}
	}
}
